py/server/deephaven/pandas.py

```diff
@@ -201,12 +201,15 @@ def _map_na(array: [np.ndarray, pd.api.extensions.ExtensionArray]):
     return array
 
 
-def to_table(df: pd.DataFrame, cols: List[str] = None) -> Table:
+def to_table(df: pd.DataFrame, cols: List[str] = None, infer_objects: bool = True) -> Table:
```

Comment:
1. I'm not a fan of the `infer_objects` variable name.  I suggest `infer_types`.  I say this because many users will not consider primitive types to be "objects".

Additionally, this name is inconsistent with the pandas method (https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.convert_dtypes.html).  In the pandas method, there are a number of arguments to control what does and doesn't get converted.  `infer_objects` is just one of the controls.  By using this parameter name for something different, it prevents us from providing these finer grain controls in the future.

2. Should we provide the finer grain controls now, or is that just too much complexity for no gain?  e.g.
```python
def to_table(df: pd.DataFrame, cols: List[str] = None, infer_types: bool = True, infer_objects=True, convert_string=True, convert_integer=True, convert_boolean=True, convert_floating=True, dtype_backend='numpy_nullable') -> Table:
```
or
```python
def to_table(df: pd.DataFrame, cols: List[str] = None, infer_objects=True, convert_string=True, convert_integer=True, convert_boolean=True, convert_floating=True, dtype_backend='numpy_nullable') -> Table:
```

Reply:
1. It is my opinion that we should not touch none-object-type columns, `infer_types` seems a bit too generic vs. `infer_objects`. 
2. `convert_dtypes()` by default not only converts object-type column but may also convert floating columns to integers ones. That's why in the implementation I only apply it to object-type columns. 
3. I am not sure adding the support of finer control over the conversion is a good idea now.  It could be more confusing than helpful.

Reply:
The fact that this is only applied to `objects` is a problem that I missed in my last look.  The problem is illustrated here:

```python

import pandas as pd
from deephaven.pandas import to_table

df = pd.DataFrame({"A":[1,2,3], "B":[1,2.1,3], "C":[1,pd.NA,3]})

t = to_table(df)
t2 = to_table(df.convert_dtypes())
```

<img width="728" alt="image" src="https://github.com/user-attachments/assets/a3bf6e2f-8588-4978-b01d-4e653df5caa0">

Reply:
Column 'C' is actually object type. With the changes in the PR, `t` and `t2` are equal.

Reply:
1. is `t3` really what a user would want? It renders `C` almost unusable.
2. the current implementation will not convert an explicitly typed primitive column, I still think `infer_objects` is more clear and appropriate.